src/main/scala/net/snowflake/spark/snowflake/catalog/SfCatalog.scala

```diff
@@ -0,0 +1,161 @@
+    checkNamespace(ident.namespace())
+    DefaultJDBCWrapper.tableExists(params, getFullTableName(ident))
+  }
+
+  override def dropTable(ident: Identifier): Boolean = {
+    checkNamespace(ident.namespace())
+    val conn = DefaultJDBCWrapper.getConnector(params)
+    conn.dropTable(getFullTableName(ident))
+  }
+
+  override def renameTable(oldIdent: Identifier, newIdent: Identifier): Unit = {
+    checkNamespace(oldIdent.namespace())
+    val conn = DefaultJDBCWrapper.getConnector(params)
+    conn.renameTable(getFullTableName(newIdent), getFullTableName(newIdent))
+  }
+
+  override def loadTable(ident: Identifier): Table = {
+    checkNamespace(ident.namespace())
+    val map = params.parameters
+    params = Parameters.mergeParameters(map +
+      (PARAM_SF_DBTABLE -> getTableName(ident)) +
+      (PARAM_SF_DATABASE -> getDatabase(ident)) +
+      (PARAM_SF_SCHEMA -> getSchema(ident)))
+    try {
+      SfTable(ident, jdbcWrapper, params)
+    } catch {
+      case _: SQLException => throw QueryCompilationErrors.noSuchTableError(ident)
+    }
+  }
+
+  override def alterTable(ident: Identifier, changes: TableChange*): Table = ???
+
+  override def namespaceExists(namespace: Array[String]): Boolean = namespace match {
+    case Array(catalog, schema) =>
+      val rs = DefaultJDBCWrapper.getConnector(params).getMetaData().getSchemas(catalog, schema)
+
+      while (rs.next()) {
+        val tableSchema = rs.getString("TABLE_SCHEM")
+        if (tableSchema == schema) return true
+      }
+      false
+    case _ => false
+  }
+
+  override def listNamespaces(): Array[Array[String]] = {
+      val schemaBuilder = ArrayBuilder.make[Array[String]]
+      val rs = DefaultJDBCWrapper.getConnector(params).getMetaData().getSchemas()
+      while (rs.next()) {
+        schemaBuilder += Array(rs.getString("TABLE_SCHEM"))
+      }
+      schemaBuilder.result
+  }
+
+  override def listNamespaces(namespace: Array[String]): Array[Array[String]] = {
+    namespace match {
+      case Array() =>
+        listNamespaces()
+      case Array(_, _) if namespaceExists(namespace) =>
+        Array()
+      case _ =>
+        throw QueryCompilationErrors.noSuchNamespaceError(namespace)
+    }
+  }
+
+  override def loadNamespaceMetadata(namespace: Array[String]): java.util.Map[String, String] = {
+    namespace match {
+      case Array(catalog, schema) =>
+        if (!namespaceExists(namespace)) {
+          throw QueryCompilationErrors.noSuchNamespaceError(Array(catalog, schema))
+        }
+        new java.util.HashMap[String, String]()
+      case _ =>
+        throw QueryCompilationErrors.noSuchNamespaceError(namespace)
+    }
+  }
+
+  override def createTable(
+                            ident: Identifier,
+                            schema: StructType,
+                            partitions: Array[Transform],
+                            properties: java.util.Map[String, String]): Table = ???
```

Comment:
`???` sounds misleading. If you think the interface functions are not applicable for Spark Connector, I just you throw an `SnowflakeConnectorFeatureNotSupportException`
This comment applied to the other similar cases.

Reply:
OK